cli/src/pcluster/templates/slurm_builder.py

```diff
@@ -328,12 +332,18 @@ def _add_slurm_compute_fleet(self):
         # terminate_compute_fleet_custom_resource.add_depends_on()
 
     def _add_private_hosted_zone(self):
-        cluster_hosted_zone = route53.CfnHostedZone(
```

Comment:
There is another place in the slurm_builder on which the cluster_dns_domain is used, in the parameter:
```python
self.cluster_dns_domain = CfnParameter(
            self.stack_scope,
            "ClusterDNSDomain",
            description="DNS Domain of the private hosted zone created within the cluster",
            default=f"{self.stack_name}.pcluster",
        )
```

Reply:
`self.cluster_dns_domain` was only used when dns domain is created by parallelcluster. Now change its default to change according to whether it is custom DNS or parallelcluster created DNS.